app/models/maintenance_tasks/run.rb

```diff
@@ -216,5 +233,19 @@ def csv_file
       return unless ActiveStorage::Attachment.table_exists?
       super
     end
+
+    # Returns a Task instance for this Run. Assigns any attributes to the Task
+    # based on the Run's parameters.
```

Comment:
It seemed to me the only place we called `run.task` was in the job, just before setting the `csv_content`. But I see now it's also called from the validation.
It just seems a bit misleading for `Run#task` to return a Task that's only half-ready to be used.
I think alternatively, there could be a method only focused on setting the task attributes based on the run (and something else would instantiate the task, maybe `valid_task_arguments?` in that case), or maybe it should take an option to skip downloading the CSV maybe so that it's more thruthful (= "this is the task instance ready for this run, but you have an option to not download the CSV").
Or we could keep it like this, Run#task gives you a Task instance that is ready, just missing CSV, and we do the remaining work in the job. We could mention it in the comment maybe.

Reply:
Ah, makes sense! I think I'd prefer to always explicitly download the CSV in the job, but I'm on board with naming `Run#task` something that indicates it's really only setting + validating the arguments :D